miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -1332,3 +1452,144 @@ end
 export.get_kernel_procedures_ptr
     push.KERNEL_PROCEDURES_PTR
 end
+
+# ACCOUNT DATA LOADERS
+# =================================================================================================
+
+#! Saves storage slots data into memory and validates that the storage commitment matches the 
+#! sequential storage hash.
+#!
+#! Inputs:
+#!   Operand stack: [STORAGE_COMMITMENT]
+#!   Advice map: {
+#!     STORAGE_COMMITMENT: [num_storage_slots, [STORAGE_SLOT_DATA]],
+#!   }
```

Comment:
Maybe we should change this to be `STORAGE_COMMITMENT: [STORAGE_SLOT_DATA]`. Then, we could use `adv.push_mapvaln` to get the number of elements and use that to calculate the number of storage slots.

Reply:
I did so when I have a dedicated procedure for the foreign account. But I had to change it during the unification with the code from the `prologue`: it turned out that we use `[num_storage_slots, [STORAGE_SLOT_DATA]]` array as a value for a storage commitment in the advice map to initialize the initial native account. 

But now I think that we should change this commitment for the native account as well, to use just a storage slot data without number of storage slots both in the prologue and in the api (my stupid habit of adapting to already written code, rather than changing it, is letting me down again).